contracts/Inventory.ligo

```diff
@@ -61,8 +74,82 @@ function update_item (const params: parameter; var storage: storage): return is
         end;
     } with ((nil: list (operation)), storage)
 
+function remove_item (const params: parameter; var storage: storage): return is
+    block {
+        const instances_map: option (item_instances) = storage.inventory[params.item_id];
+
+        case instances_map of
+            None -> failwith ("NO_SUCH_ITEM_IN_INVENTORY")
+            | Some (im) -> block {
+                const instance: option (item_data) = im [params.instance_number];
+
+                case instance of
+                    None -> failwith("NO_SUCH_INSTANCE_NUMBER")
+                |   Some (i) -> block {
+                    ignore_item_data (i);
+                      const removed_im : item_instances = Map.remove (params.instance_number, im);
+                      storage.inventory [params.item_id] := removed_im
+                }
+                end;
+            }
+        end;
+    } with ((nil: list (operation)), storage)
+
+function transfer_item (const params: transfer_parameter; var storage: storage): return is
+    block {
+
+        var ops : list (operation) := nil;
+        const instances_map: option (item_instances) = storage.inventory[params.item_id];
+
+        case instances_map of
+            None -> failwith ("NO_SUCH_ITEM_IN_INVENTORY")
+            | Some (im) -> block {
+
+                const instance: option (item_data) = im [params.instance_number];
+
+                case instance of
+                    None -> failwith("NO_SUCH_INSTANCE_NUMBER")
+                |   Some (i) -> block {
+
+                    const inventory : contract (remove_parameter) =
+                        case (Tezos.get_entrypoint_opt ("%remove_item_proxy", Tezos.self_address) : option (contract (remove_parameter))) of
+                            Some (contract) -> contract
+                            | None -> (failwith ("CONTRACT_NOT_FOUND") : contract (remove_parameter))
+                        end;
+
+                    const remove_action : remove_parameter = Remove_item (record [
+                        data = i;
```

Comment:
we don't need to specify data to remove an item, passing instance_number and item_id should be enough.
Feel free to create a new parameter type for remove_item that excludes data. It can be a tuple like:

`remove_item_parameter is nat * nat;`